src/VecSim/algorithms/hnsw/hnsw_tiered.h

```diff
@@ -522,8 +522,17 @@ size_t TieredHNSWIndex<DataType, DistType>::indexLabelCount() const {
 template <typename DataType, typename DistType>
 int TieredHNSWIndex<DataType, DistType>::addVector(const void *blob, labelType label,
                                                    void *auxiliaryCtx) {
-    /* Note: this currently doesn't support overriding (assuming that the label doesn't exist)! */
+    int ret = 1;
     this->flatIndexGuard.lock();
+    if (this->frontendIndex->isLabelExists(label) && !this->backendIndex->isMultiValue()) {
+        // Overwrite the vector and invalidate its only pending job (since we are not in MULTI).
+        this->frontendIndex->deleteVector(label);
+        auto *old_job = this->labelToInsertJobs.at(label).at(0);
+        old_job->id = INVALID_JOB_ID;
```

Comment:
`this->flatIndexGuard` is the guard for reading this flag in the execution job as well, right?

Reply:
yes